apps/src/studio/studio.js

```diff
@@ -1851,6 +1851,14 @@ Studio.checkForPreExecutionFailure = function () {
     return true;
   }
 
+  if (studioApp.hasEmptyFunctionOrVariableName()) {
+    Studio.result = false;
+    Studio.testResults = TestResults.EMPTY_FUNCTION_NAME;
+    Studio.message = commonMsg.unnamedFunction();
+    Studio.preExecutionFailure = true;
+    return true;
+  }
```

Comment:
We don't really have a great common place to put this sort of thing. Our feedback code in general is kind of a mess. Also, these are the only three apps (at least currently) that will ever have functional_definitions, since other apps dont use the functional style